src/filemenu/filemenu_common.c

```diff
@@ -717,6 +778,104 @@ void filemenu_update_hidden_name_confirm(
     }
 }
 
+#if VERSION_PAL
+void filemenu_update_pal_80247f40(
+    s32 windowIndex,
+    s32* flags,
+    s32* posX, s32* posY, s32* posZ,
+    f32* scaleX, f32* scaleY,
+    f32* rotX, f32* rotY, f32* rotZ,
+    s32* darkening,
+    s32* opacity
+) {
+    Window* window = &gWindows[windowIndex];
+    s32 var_a1 = window->updateCounter;
+    s32 var_a3 = 0;
+    s32 var_v1;
+
+    switch (windowIndex) {
+        case WINDOW_ID_FILEMENU_FILE0_INFO:
+            var_a3 = 0;
+            break;
+        case WINDOW_ID_FILEMENU_FILE1_INFO:
+            var_a3 = 1;
+            break;
+        case WINDOW_ID_FILEMENU_FILE2_INFO:
+            var_a3 = 2;
+            break;
+        case WINDOW_ID_FILEMENU_FILE3_INFO:
+            var_a3 = 3;
+            break;
+    }
+    var_v1 = gCurrentLanguage;
+    var_v1 -= var_a3;
+    var_v1 = abs(var_v1);
+
+    var_a1 -= (var_v1 * 3);
+    if (var_a1 < 0) {
+        var_a1 = 0;
+    }
+    if (var_a1 == 0) {
+        window->flags &= ~WINDOW_FLAG_HIDDEN;
+    }
+    if (var_a1 < ARRAY_COUNT(D_filemenu_8024EB5C)) {
+        *posX -= D_filemenu_8024EB5C[var_a1];
+    } else {
+        *posX -= D_filemenu_8024EB5C[ARRAY_COUNT(D_filemenu_8024EB5C) - 1];
+        window->flags &= ~WINDOW_FLAG_INITIAL_ANIMATION;
+    }
+}
+
+void filemenu_pal_80248018(
+    s32 windowIndex,
+    s32* flags,
+    s32* posX, s32* posY, s32* posZ,
+    f32* scaleX, f32* scaleY,
+    f32* rotX, f32* rotY, f32* rotZ,
+    s32* darkening,
+    s32* opacity
+) {
+    Window* window = &gWindows[windowIndex];
+    s32 var_a1 = window->updateCounter;
+    s32 var_a3 = 0;
+    s32 var_v1;
+
+    switch (windowIndex) {
+        case WINDOW_ID_FILEMENU_FILE0_INFO:
+            var_a3 = 0;
+            break;
+        case WINDOW_ID_FILEMENU_FILE1_INFO:
+            var_a3 = 1;
+            break;
+        case WINDOW_ID_FILEMENU_FILE2_INFO:
+            var_a3 = 2;
+            break;
+        case WINDOW_ID_FILEMENU_FILE3_INFO:
+            var_a3 = 3;
+            break;
+    }
+    var_v1 = gCurrentLanguage;
+    var_v1 -= var_a3;
+    if (var_v1 < 0) {
+        var_v1 = -var_v1;
+    }
+    var_a1 -= (var_v1 * 3);
+    if (var_a1 < 0) {
+        var_a1 = 0;
+    }
+    if (var_a1 < ARRAY_COUNT(D_filemenu_8024EB80)) {
+        *posX -= D_filemenu_8024EB80[var_a1];
+    } else {
+        *posX -= D_filemenu_8024EB80[ARRAY_COUNT(D_filemenu_8024EB80) - 1];
+        window->flags &= ~WINDOW_FLAG_INITIAL_ANIMATION;
+        window->flags |= WINDOW_FLAG_HIDDEN;
+    }
+    if (var_a1 >= 0x10) {
```

Comment:
Do we need to express this in relation to `ARRAY_COUNT(D_filemenu_8024EB80)`?

Reply:
I looked at that, but I don't think I can really tell without more context. it's like array count minus 1 or 2 right?

Reply:
Best to leave `// TODO: ARRAY_COUNT?` on this one I think